core/src/main/java/nl/nn/adapterframework/doc/model/FrankElement.java

```diff
@@ -1,27 +1,231 @@
+/* 
+Copyright 2020 WeAreFrank! 
+
+Licensed under the Apache License, Version 2.0 (the "License"); 
+you may not use this file except in compliance with the License. 
+You may obtain a copy of the License at 
+
+    http://www.apache.org/licenses/LICENSE-2.0 
+
+Unless required by applicable law or agreed to in writing, software 
+distributed under the License is distributed on an "AS IS" BASIS, 
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+See the License for the specific language governing permissions and 
+limitations under the License. 
+*/
+
 package nl.nn.adapterframework.doc.model;
 
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashMap;
 import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.Logger;
 
 import lombok.Getter;
-import lombok.Setter;
+import nl.nn.adapterframework.doc.DocWriterNew;
+import nl.nn.adapterframework.doc.Utils;
+import nl.nn.adapterframework.util.LogUtil;
 
 public class FrankElement {
+	private static Logger log = LogUtil.getLogger(FrankElement.class);
+
 	private final @Getter String fullName;
 	private final @Getter String simpleName;
-	private @Getter @Setter FrankElement parent;
-	private @Getter @Setter List<FrankAttribute> attributes;
-	private @Getter @Setter List<ConfigChild> configChildren;
+	private final @Getter boolean isAbstract;
+
+	// Represents the Java superclass.
+	private @Getter FrankElement parent;
+
+	private LinkedHashMap<String, FrankAttribute> attributes;
+	private LinkedHashMap<ConfigChild.Key, ConfigChild> configChildren;
+	private @Getter List<ConfigChild> aliasSources;
+	private String cachedAlias = null;
+	private @Getter FrankElementStatistics statistics;
 
 	FrankElement(Class<?> clazz) {
-		this(clazz.getName(), clazz.getSimpleName());
+		this(clazz.getName(), clazz.getSimpleName(), Modifier.isAbstract(clazz.getModifiers()));
 	}
 
 	/**
 	 * Constructor for testing purposes. We want to test attribute construction in isolation,
 	 * in which case we do not have a parent.
+	 * TODO: Reorganize files such that this test constructor need not be public.
 	 */
-	FrankElement(final String fullName, final String simpleName) {
+	public FrankElement(final String fullName, final String simpleName, boolean isAbstract) {
 		this.fullName = fullName;
 		this.simpleName = simpleName;
+		this.isAbstract = isAbstract;
+		this.aliasSources = new ArrayList<>();
+	}
+
+	public FrankElement(final String fullName, final String simpleName) {
+		this(fullName, simpleName, false);
+	}
+	
+	public void setParent(FrankElement parent) {
+		this.parent = parent;
+		this.statistics = new FrankElementStatistics(this);
+	}
+
+	/**
+	 * Setter for attributes. We prevent modifying the list of attributes
+	 * because we want to maintain the private field attributeLookup.
```

Comment:
De reden om de attributes niet te kunnen modificeren is er niet meer. Of is er nog een andere reden?

Reply:
Ik heb dit comment weggehaald.

Reply:
Dit overbodige commentaar stond er ook voor config children. Daar is het nu ook weg.